src/Symfony/Bundle/FrameworkBundle/Controller/RedirectController.php

```diff
@@ -51,7 +51,7 @@ public function redirectAction(Request $request, $route, $permanent = false, $ig
 
         $attributes = array();
         if (false === $ignoreAttributes || is_array($ignoreAttributes)) {
-            $attributes = $request->attributes->get('_route_params');
+            $attributes = array_merge($request->attributes->get('_route_params'), $request->query->all());
```

Comment:
If this were accepted, I think the implementation should match what's done in `urlRedirectAction` - i.e. don't merge the query params into the attributes, but instead append the `?` or `&` at the end with the `$request->getQueryString()` method return.